components/formats-bsd/src/loci/formats/tiff/TiffSaver.java

```diff
@@ -759,14 +764,52 @@ public void overwriteIFDValue(RandomAccessInputStream raf,
       throw new FormatException(
         "No such IFD (" + ifd + " of " + offsets.length + ")");
     }
-    raf.seek(offsets[ifd]);
+    overwriteIFDValue(raf, offsets[ifd], tag, value);
+  }
+
+  /**
+   * Surgically overwrites an existing IFD value with the given one. This
+   * method requires that the IFD directory entry already exist. It
+   * intelligently updates the count field of the entry to match the new
+   * length. If the new length is longer than the old length, it appends the
+   * new data to the end of the file and updates the offset field; if not, or
+   * if the old data is already at the end of the file, it overwrites the old
+   * data in place.
+   *
+   * @param raf the input stream representing the file to be edited
+   * @param ifdOffset the offset to the IFD
+   * @param tag the tag code
+   * @param value the new value for the tag
+   */
+  public void overwriteIFDValue(RandomAccessInputStream raf,
+    long ifdOffset, int tag, Object value) throws FormatException, IOException
+  {
+    raf.seek(0);
```

Comment:
With this refactoring, does that mean any call to the current `overwriteIFDValue(RandomAccessInputStream, int, int, Object)` API will now perform a redundant check on the first IFD?

Reply:
The header will be checked twice, yes.  Could refactor further to add a ```overwriteIFDValue(RandomAccessInputStream, long, int, Object, boolean skipHeaderCheck)```?  Otherwise not sure of a good and safe way to skip that check.

Reply:
Since we're introducing new API anyways, I think introducing 

` public void overwriteIFDValue(RandomAccessInputStream raf,
    long ifdOffset, int tag, Object value, boolean skipskipHeaderCheck)`

is acceptable. From there both `public void overwriteIFDValue(RandomAccessInputStream raf,
    long ifdOffset, int tag, Object value)` and `public void overwriteIFDValue(RandomAccessInputStream raf,
    int ifd, int tag, Object value)` can delegate to this API by passing the appropriate boolean value.

Reply:
Added in b43e748